internal/metrics/metrics.go

```diff
@@ -0,0 +1,28 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	OpenstackConnectionMetric = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "external_dns_webhook_openstack_connection",
+		Help: "Indicates if the webhook has a connection to the OpenStack API (1 for connected, 0 for not connected)",
```

Comment:
This indicates that the webhook hold open a permanent connection to openstack, I don't think that's true though.
Since the metric only is set once during startup, maybe it would make more sense to call it `external_dns_webhook_openstack_connection_initialized` and update the Help-Text to reference that.

Otherwise, we should update the metric whenever one of our API calls fails due to connection reasons, or after an API is successful, but in that case I believe the metric would be pretty redundant with `external_dns_webhook_failed_api_calls_total` and `external_dns_webhook_total_api_calls`.

What is your intention with this metric?